main.tf

```diff
@@ -68,7 +69,7 @@ resource "aws_db_proxy_target" "db_cluster" {
 }
 
 resource "aws_db_proxy_endpoint" "this" {
-  for_each = var.db_proxy_endpoints
+  for_each = local.db_proxy_endpoints
```

Comment:
```suggestion
  for_each = {for k,v in var.db_proxy_endpoints: k => v if var.create_proxy }
```

Reply:
I did try this first, but I couldn't get it to work... 

I'll try again though...

Reply:
And for whatever reason it worked this time... Updated in https://github.com/fatmcgav-depop/terraform-aws-rds-proxy/commit/2a76143c5cd33b029653dc2d07efc3d46b459d93